prime-router/src/main/kotlin/Mappers.kt

```diff
@@ -742,6 +742,60 @@ class NullMapper : Mapper {
     }
 }
 
+/**
+ * Maps a provided code ID and coding system to a verified code in a valueset containined in the [metadata]
+ * using the target element's valueset. It also performs some convertions.  A valueSet is required for the
+ * element using this mapper, otherwise null is returned.
+ */
+class CodedElementMapper(val metadata: Metadata) : Mapper {
+    override val name = "codedElement"
+
+    override fun valueNames(element: Element, args: List<String>): List<String> {
+        if (args.size != 2) error("$name mapper requires two arguments: source code ID and coding system")
+        if (element.valueSet.isNullOrBlank()) error("Element must have a value set to use the $name mapper")
+        return args
+    }
+
+    override fun apply(element: Element, args: List<String>, values: List<ElementAndValue>): String? {
+        return if (values.isEmpty() || values.size != 2) {
+            null
+        } else {
+            var value: String? = null
+            if (!element.valueSet.isNullOrBlank()) {
+                val system = values[1].value
+                val verifiedCode = element.toCode(values[0].value, metadata.findValueSet(system.lowercase()))
+                value = when {
+                    // Fix incorrect codes for hl70189
+                    "hl70189".equals(element.valueSet, true) && "hl70189".equals(system, true) -> {
```

Comment:
Hmm.   I think I hit a very similar problem a few months back.  My hack solution was different from yours, as found in direct/direct-covid-19.schema:

```
  - name: patient_ethnicity
    documentation:  Internally, ReportStream uses hl70189 (H,N,U), but should use HHS values. (2135-2, 2186-5, UNK, ASKU). A mapping is done here, but\
 better is to switch all of RS to HHS standard.
    altValues:
      - code: H
        display: 2135-2
      - code: N
        display: 2186-5
      - code: U
        display: UNK
      - code: U
        display: ASKU
    csvFields: [{ name: patientEthnicity, format: $alt}]
```

Reply:
Removed these changes as discussed since it needs a lot more thought behind it.  Will leave for #1468